Shifty.App/Components/Refunds.razor

```diff
@@ -0,0 +1,104 @@
+@using Shifty.Api.Generated.AnalogCoreV2
+@using Shared
+@using LanguageExt.UnsafeValueAccess
+@using Components
+@using Shifty.App.DomainModels
+@using System.Collections.ObjectModel
+@inject IPurchaseService PurchaseService
+@inject ISnackbar Snackbar
+
+<MudPaper Elevation="15" Style="border-radius: 5px; width:fit-content; min-width: 500px; margin:2rem auto">
+            @if(_purchasesFetched)
+            {
+                <MudIconButton Icon="@Icons.Material.Filled.ArrowBack" Color="Color.Primary" OnClick="@(() => _purchasesFetched = false)" />
+                <MudDataGrid
+                    T="Purchase"
+                    Items="@_purchases"
+                    EditMode="DataGridEditMode.Form"
+                    ReadOnly="false"
+                    FixedHeader="true"
+                    Height="calc(100vh - 250px)"
+                    SortMode="MudBlazor.SortMode.None"
+                    Style="width: 70vw;padding: 1rem"
+                >
+                    <Columns>
+                        <PropertyColumn Property="x => x.Id" Title="Id" IsEditable="false" />
+                        <PropertyColumn Property="x => x.DateCreated" Title="Date" IsEditable="false"/>
+                        <PropertyColumn Property="x => x.ProductName" Title="Name" IsEditable="false" />
+                        <PropertyColumn Property="x => x.NumberOfTickets" Title="Tickets" IsEditable="false" />
+                        <PropertyColumn Property="x => x.Price" Title="Price" IsEditable="false" />
+                        <PropertyColumn Property="x => x.PurchaseStatus" Title="Status" IsEditable="false">
+                            <CellTemplate>
+                            @{
+                                if(@context.Item.PurchaseStatus == PurchaseStatus.Completed)
+                                {
+                                    <MudButton Size="@Size.Small" Color="Color.Primary" Variant="Variant.Filled" OnClick="@(() => RefundPurchase(context.Item))">
+                                        Refund
+                                    </MudButton>
+                        }
+                        else
+                        {
+                            <MudText>@context.Item.PurchaseStatus</MudText>
+                        }
+                            }
+                            </CellTemplate>
+                        </PropertyColumn>
+                    </Columns>
+                </MudDataGrid>
+            }
+            else
+            {
+                <MudContainer Style="padding: 2rem; display: flex; flex-direction: column; justify-content: center;">
+                    <MudText Style="padding: 0.5rem;">Find purchases for user</MudText>
+                    <MudTextField Immediate="true" OnKeyDown="OnKeyDown" @bind-Value="UserId" Label="User ID" Variant="Variant.Outlined" Adornment="Adornment.End" AdornmentIcon="@Icons.Material.Filled.Search" AdornmentColor="Color.Secondary"></MudTextField>
```

Comment:
Would it make sense to add a `Mask` here to prevent inserting letters?

[MudBlazor pattern masking](https://mudblazor.com/features/masking#patternmask)